plugins/modules/ipadnsforwardzone.py

```diff
@@ -0,0 +1,246 @@
+            # general
+            ipaadmin_principal=dict(type="str", default="admin"),
+            ipaadmin_password=dict(type="str", required=False, no_log=True),
+
+            name=dict(type="str", aliases=["cn"], default=None,
+                      required=True),
+            forwarders=dict(type='list', aliases=["idnsforwarders"],
+                            required=False),
+            forwardpolicy=dict(type='str', aliases=["idnsforwardpolicy"],
+                               required=False,
+                               choices=['only', 'first', 'none']),
+            skip_overlap_check=dict(type='bool', required=False),
+            action=dict(type="str", default="dnsforwardzone",
+                        choices=["member", "dnsforwardzone"]),
+            # state
+            state=dict(type='str', aliases=['ensure'], default='present',
+                       choices=['present', 'absent']),
+        ),
+        supports_check_mode=True,
+    )
+
+    ansible_module._ansible_debug = True
+
+    # Get parameters
+
+    # general
+    ipaadmin_principal = module_params_get(ansible_module,
+                                           "ipaadmin_principal")
+    ipaadmin_password = module_params_get(ansible_module,
+                                          "ipaadmin_password")
+    name = module_params_get(ansible_module, "name")
+    action = module_params_get(ansible_module, "action")
+
+    # present
+    forwarders = module_params_get(ansible_module, "forwarders")
+    forwardpolicy = module_params_get(ansible_module, "forwardpolicy")
+    skip_overlap_check = module_params_get(ansible_module,
+                                           "skip_overlap_check")
+
+    # state
+    state = module_params_get(ansible_module, "state")
+
+    # Check parameters
+    if state == "absent":
+        invalid = ["forwarders", "forwardpolicy", "skip_overlap_check"]
+        for x in invalid:
+            if vars()[x] is not None:
+                ansible_module.fail_json(
+                    msg="Argument '%s' can not be used with action "
+                    "'%s'" % (x, action))
+
+    if state == "present":
+        required = ["forwarders"]
+        for x in required:
+            if vars()[x] is None:
+                ansible_module.fail_json(
+                    msg="Argument '%s' is required when state=present "
+                    "'%s'" % (x, action))
+
+    # Init
+
+    changed = False
+    exit_args = {}
+    args = {}
+    ccache_dir = None
+    ccache_name = None
+    try:
+        if not valid_creds(ansible_module, ipaadmin_principal):
+            ccache_dir, ccache_name = temp_kinit(ipaadmin_principal,
+                                                 ipaadmin_password)
+        api_connect()
+
+        # Make sure forwardzone exists
+        res_find = find_dnsforwardzone(ansible_module, name)
+        if action == "member":
+            forwarders = list(set(res_find["idnsforwarders"] + forwarders))
```

Comment:
With this line is not possible to make sure that entries are not part of idnsforwarders (`action: member` with `state: absent`). Also res_find is used without checking that it is not None.

Reply:
Please also add a test case to ensure the absence of an entry with also testing that changed is false for the repeated task.